code-challenges.test.js

```diff
@@ -14,33 +14,329 @@
 
-// a) Create a test with an expect statement using the variables provided.
+// ● createSent › takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+//   ReferenceError: createSent is not defined
 
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+// const createSent = (array) => {
 
-// a) Create a test with an expect statement using the variables provided.
+//   // Iterates through the array
+//   array.map(value => {
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+//     // Assigns current value's name values to tempArr
+//     let tempArr = value.name.split(" ")
 
-// b) Create the function that makes the test pass.
+//     // Iterates through tempArry
+//     for (let i = 0; i < tempArr.length; i++) {
+
+//       // Changes first letter to a capital
+//       tempArr[i] = tempArr[i].charAt(0).toUpperCase() + tempArr[i].slice(1)
+//     }
+
+//     // Replaces current value's name values with the new uppercased ones in 'tempArr'
+//     value.name = tempArr.join(" ")
+
+//   })
+
+//   // Iterates through the given array again
+//   return array.map(value => {
+
+//     // Returns the requested sentence
+//     return `${value.name} is ${value.occupation}.`
+
+//   })
+// }
+
+// Success 
+// PASS  ./challenges.test.js
+  // createSent
+  // ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (2 ms)
+  
+  
+// Refactor 1
+// Pseudo Code:
+  // Process:
+    // Remove temporary array
+    // Remove nested for loop
+    // Use value.name to access 'name' key values
+    // Append .split() to create new array
+    // Append .map to iterate over newly created array
+      // Return the names with uppercased first letters in a sentence
+  // Expected Output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+const createSent = (array) => {
+
+  // Iterates through given array
+  return array.map(value => {
+
+    // .name accessess 'name' field of each object in array
+    // .split() creates array of the 'name' field
+    // .map() iterates through that array
+    return value.name.split(" ").map(value => 
+
+      // .charAt & .toUpperCase uppercase the first letter in each word
+      // .slice appends the rest of the word to the uppercased letter
+      // .join returns array to original single string
+      value.charAt(0).toUpperCase() + value.slice(1)).join(" ") + " is " + value.occupation + "."
```

Comment:
Good refactor and pseudo code. I suggest keeping the pseudo code either all above or all inline